examples/benchmark/src/main.c

```diff
@@ -0,0 +1,150 @@
+#include <mpix/image.h>
+#include <mpix/low_level.h>
+#include <mpix/port.h>
+#include <mpix/print.h>
+#include <mpix/types.h>
+#include <mpix/utils.h>
+#include <mpix/stats.h>
+
+#ifndef CONFIG_WIDTH
+#define CONFIG_WIDTH 320
+#endif
+
+#ifndef CONFIG_HEIGHT
+#define CONFIG_HEIGHT 240
+#endif
+
+#define WIDTH  CONFIG_WIDTH
+#define HEIGHT CONFIG_HEIGHT
+
+#define CHECK(x)                                                                                   \
+	({                                                                                         \
+		int e = (x);                                                                       \
+		if (e) {                                                                           \
+			fprintf(stderr, "%s: %s\n", #x, strerror(-e));                             \
+			return e;                                                                  \
+		}                                                                                  \
+	})
+
+uint8_t srcbuf[WIDTH * HEIGHT * 3];
+uint8_t dstbuf[WIDTH * HEIGHT * 3];
+
+#define NUM_OPS_RUN 3
+
+struct benchmark_desc {
+	const char *name;
+	enum mpix_op_type type;
+	const int32_t *params;
+	size_t params_nb;
+	uint32_t fourcc;
+};
+
+#define BENCHMARK_DESC_AUTO(_name, _type, _fourcc, ...)                                            \
+	(struct benchmark_desc)                                                                    \
+	{                                                                                          \
+		.name = (_name), .type = (_type), .params = (const int32_t[]){__VA_ARGS__},        \
+		.params_nb = ((size_t)(sizeof((const int32_t[]){__VA_ARGS__}) / sizeof(int32_t))), \
+		.fourcc = (_fourcc),                                                               \
+	}
```

Comment:
You can get the name from the operation itself after it was added, or using `#_type`, to reduce the number of fields even more. Then it just becomes `_fourcc, _type, _params`.

Reply:
ah, right, there are multiple names depending on the parameters...

it's probably fine as it is then, thanks!